lib/commands/plugin-test.sh

```diff
@@ -1,69 +1,68 @@
 fail_test() {
     echo "FAILED: $1"
-    rm -rf $ASDF_DIR
+    rm -rf "$ASDF_DIR"
     exit 1
 }
 
 plugin_test_command() {
-    export ASDF_DIR=$(mktemp -dt asdf.XXXX)
-    git clone https://github.com/asdf-vm/asdf.git $ASDF_DIR
+    ASDF_DIR=$(mktemp -dt asdf.XXXX)
+    export ASDF_DIR
+    git clone https://github.com/asdf-vm/asdf.git "$ASDF_DIR"
 
     local plugin_name=$1
     local plugin_url=$2
-    local plugin_command="${@:3}"
+    local plugin_command="${*:3}"
 
-    if [ -z "$plugin_name" -o -z "$plugin_url" ]; then
+    if [ -z "$plugin_name" ] || [ -z "$plugin_url" ]; then
         fail_test "please provide a plugin name and url"
     fi
 
-    (asdf plugin-add $plugin_name $plugin_url)
-    if [ $? -ne 0 ]; then
+    if ! (asdf plugin-add "$plugin_name" "$plugin_url"); then
         fail_test "could not install $plugin_name from $plugin_url"
     fi
 
-    if ! asdf plugin-list | grep $plugin_name > /dev/null; then
+    if ! (asdf plugin-list | grep "$plugin_name" > /dev/null); then
         fail_test "$plugin_name was not properly installed"
     fi
 
-    read -a versions <<< $(asdf list-all $plugin_name)
-
-    if [ $? -ne 0 ]; then
+    local versions
+    # shellcheck disable=SC2046
+    if ! read -r -a versions <<< $(asdf list-all "$plugin_name"); then
         fail_test "list-all exited with an error"
     fi
 
     if [ ${#versions} -eq 0 ]; then
         fail_test "list-all did not return any version"
     fi
 
+    local latest_version
     latest_version=${versions[${#versions[@]} - 1]}
 
-    (asdf install $plugin_name $latest_version)
-
-    if [ $? -ne 0 ]; then
+    if ! (asdf install "$plugin_name" "$latest_version"); then
         fail_test "install exited with an error"
     fi
 
-    cd $ASDF_DIR
-    (asdf local $plugin_name $latest_version)
-    if [ $? -ne 0 ]; then
+    cd "$ASDF_DIR" || fail_test "could not cd $ASDF_DIR"
+
+    if ! (asdf local "$plugin_name" "$latest_version"); then
         fail_test "install did not add the requested version"
     fi
 
-    (asdf reshim $plugin_name)
-    if [ $? -ne 0 ]; then
+    if ! (asdf reshim "$plugin_name"); then
         fail_test "could not reshim plugin"
     fi
 
     if [ -n "$plugin_command" ]; then
-        (PATH="$ASDF_DIR/bin":"$ASDF_DIR/shims":$PATH eval "$plugin_command")
+        (PATH="$ASDF_DIR/bin:$ASDF_DIR/shims:$PATH" eval "$plugin_command")
+        local exit_code
         exit_code=$?
-        if [ $exit_code -ne 0 ]; then
```

Comment:
Did shellcheck suggest this? I was thinking `-ne` was a more widely supported condition.

Reply:
Yea, I didn't make any changes that were not suggested by shellcheck. If they are syntactically correct I would tend to error on the side of shellcheck and there may be more info about why it was suggested on the rule definition doc page.